src/main/java/spoon/reflect/code/CtTry.java

```diff
@@ -47,6 +47,19 @@ public interface CtTry extends CtStatement, TemplateParameter<Void>, CtBodyHolde
 	@PropertySetter(role = CATCH)
 	<T extends CtTry> T addCatcher(CtCatch catcher);
 
+	/**
+	 * Adds a catch block at the specified position in the <code>try</code> statement.
+	 * Behaves similarly to {@link java.util.List#add(int, Object)}.
+	 *
+	 * @param <T> the type of the try statement
+	 * @param position the position at which the <code>catcher</code> is to be inserted
+	 * @param catcher the catch statement to be inserted
+	 * @return this try statement
+	 * @throws IndexOutOfBoundsException if the position is out of range (position < 0 || position > number of catchers)
+	 */
+	@PropertySetter(role = CATCH)
+	<T extends CtTry> T addCatcherAt(int position, CtCatch catcher);
```

Comment:
I am using `position` to refer to `index` in the `catchers` collection because it has been used for all analogous methods. See https://github.com/INRIA/spoon/blob/76ffd3353a535b0ce6edf0bf961a05236a40d3a1/src/main/java/spoon/reflect/code/CtAbstractSwitch.java#L72 as an example.